### ISSUE TITLE
getaddrmaninfo RPC: add Transport v1/v2 to `tried` for ipv4 & ipv6

### ISSUE THREAD

**Issue by MarnixCroes** [bitcoin/bitcoin#28807]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behaviour

Would be nice if the transport version is added to `getaddrmaninfo` RPC for ipv4 and ipv6, so the user can see how many Transport v1/v2 one's node has connected to.
current:

`getaddrmaninfo`

{
  "ipv4": {
    "new": x,
    "tried": x,
    "total": x
  },
  "ipv6": {
    "new": x,
    "tried": x,
    "total": x
  },
  "onion": {
    "new": x,
    "tried": x,
    "total": x
...

### Expected behaviour

`getaddrmaninfo`

{
  "ipv4": {
    "new": x,
    "tried": x, {
      "transportv1": x,
      "transportv2": x, }
    "total": x
  },
  "ipv6": {
    "new": x,
    "tried": x, {
      "transportv1": x,
      "transportv2": x, }
    "total": x
  },
  "onion": {
    "new": x,
    "tried": x,
    "total": x
...

or something like this

### Steps to reproduce

.

### How did you obtain Bitcoin Core

Compiled from source

### What version of Bitcoin Core are you using?

v26.0.rc2

### Operating system and version

.

**Comment by mzumsande** [comment]
I think that this is not so straightforward because we don't save historic connection details in addrman (and I'm not sure if we want to start doing that). What we do save is the service flags , which could be dumped with the debug-only `getrawaddrman` rpc.

However, a BIP324 service flag for an addr in `tried` does not necessarily indicate that we connected to them via v2. The service flag may just be wrong (and we connected to them but downgraded the conection to v1), or maybe it is true now but either we or them did not run with `-v2transport` at the time the connection was made.

**Comment by MarnixCroes** [comment]
@mzumsande OK.
Yes I was thinking maybe adding something like `gettransportinfo`/`getclearnettransportinfo` might be a good addition/alternative to this one.
Where detailed info, like you mentioned, could be shown.

**Comment by brunoerg** [comment]
I agree with @mzumsande. Also, I'm not sure but I do believe you could use tracepoints to track it. See: https://github.com/bitcoin/bitcoin/blob/master/doc/tracing.md

**Comment by willcl-ark** [comment]
This issue hasn’t attracted much interest from other contributors in quite some time.

Given that, it doesn’t seem important enough to keep open indefinitely. I’m going to close it for now due to lack of activity, but pull requests or renewed discussion are always welcome.

Comment here if you think this shoudl be re-opened.
